DRAFT.php

```diff
@@ -35,17 +37,17 @@
  * is undefined, or to avoid ambiguous results. This is a marker interface to
  * support catching all access exceptions.
  */
-interface AccessException {}
+interface AccessException extends Throwable {}
 
 /**
  * Marker interface for all state-related exceptions, such as heap corruption.
  */
-interface StateException {}
+interface StateException extends Throwable {}
 
 /**
  * Should be thrown when an empty container is accessed a clear, obvious result.
  */
-class EmptyStateException implements StateException {}
+class EmptyStateException extends LogicException implements StateException {}
```

Comment:
I think we should use `RuntimeException` consistently here.

Reply:
Or change the others to `LogicException` also.

Reply:
Not a strong opinion here, although _obvious result_ in the description suggests a logic error. :)

Reply:
This is quite old but... http://ralphschindler.com/2010/09/15/exception-best-practices-in-php-5-3

> If you look at the anatomy of an object, one of two things is generally happening. First, the object will be tracking and mutating state. This means the object is generally not doing anything (yet); it might have configuration passed to it; it might be setting up properties (via setters and getters); or, it might be getting references to other objects. Second, when the object is not tracking and mutating state, it is operating – doing what it was designed to do. This is the object’s runtime. For instance, during the objects lifetime, it might be created, passed a configure object, then it might have setFoo($foo), setBar($bar) called. During these times any kind of LogicException should be raised. In addition, when the object is asked to do something, with parameters, for example $object->doSomething($someVariation); during the first few lines when it interrogates that $someVariation variable, it would throw a LogicException. After it is done interrogating $someVariation, and it goes on about doing its job of doSomething(), this is considered its “runtime” and in this code it would throw RuntimeExcpetions.